### DISCUSSION TITLE
Python bindings

### DISCUSSION THREAD

**Discussion by absadiki** [ggml-org/whisper.cpp#603] [Bindings]
Hello @ggerganov,

First, the project is really awesome, great job :clap: 

I was developing [this](https://github.com/abdeladim-s/subsai) and the past week I came across your project, I didn't find any easy to use Python bindings that fully support the C-Style `whisper.h` API yet, so I thought it would be fun to learn and create one.

So I wanted to share with you, and the community, the results:
The bindings source code is here: [pywhispercpp](https://github.com/abdeladim-s/pywhispercpp) and the documentation is hosted [here](https://abdeladim-s.github.io/pywhispercpp/). 
The package supports almost all `whisper.h` functions, plus I built an easy to use ('Pythonic') API on top of them.
I pushed the wheels to [pypi](https://pypi.org/project/pywhispercpp/) as awell. 

Hope it will be useful for the community.

**Comment by HowardMcfalllx** [comment]
Hello! It's great to hear that you've been enjoying @ggerganov's project and that you've taken the initiative to create Python bindings for the C-Style whisper.h API.

Your pywhispercpp bindings and documentation sound like a great contribution to the community, especially since you've built an easy-to-use API on top of the existing functions. It's also great to hear that you've uploaded the package to PyPI for easy installation.

Contributions like yours can make a big difference in the accessibility and usability of open-source projects, and it's great to see members of the community supporting each other in this way.

Thank you for sharing your work with the community, and we wish you the best of luck in your future contributions!

**Reply by absadiki** [reply]
My pleasure .. 'Glad you find it useful.
Thanks @HowardMcfalllx

**Comment by zac-wang-nv** [comment]
Hi,
Thanks for this great contribution!

have you benchmarked the speed of the python binding implementation
compared to the original python implementation of whisper yet?

**Reply by absadiki** [reply]
Hi, 
You are welcome.

No honestly I didn't, because generally, the bindings are just a mapping to the `C++` functions of `whisper.cpp`, so the performance should be almost the same as `whisper.cpp`.

@ggerganov has already mentioned the difference between the two implementations [here](https://github.com/ggerganov/whisper.cpp/issues/430#issuecomment-1398858842) : 

> Hi! Whisper is the original speech recognition model created and released by OpenAI. It is implemented in Python and supports running both on the CPU and on the GPU.
> 
> whisper.cpp is a custom inference implementation of the same model. It's implemented in C/C++ and runs only on the CPU. It tries to provide the same functionality as the original, but there are differences in the implementation.
> 
> In terms of speed, Whisper running on a GPU would in general be more efficient than the other options - multiple times faster on modern GPUs. However, if you want to run the model on a CPU, in some cases whisper.cpp can give you advantage. For example, currently on Apple Silicon, whisper.cpp should be faster.
> 
> In terms of accuracy, Whisper is the "gold standard". whisper.cpp should be similar and sometimes worse.
> 
> More information is available in the [F.A.Q. discussion](https://github.com/ggerganov/whisper.cpp/discussions/126).

 
Personally, I find `whisper.cpp` really good on the CPU compared to the original whisper. 
You can find benchmark results from other people in this [issue](https://github.com/ggerganov/whisper.cpp/issues/89) as well.

You can also use the[ web-ui](https://github.com/abdeladim-s/subsai) I was working on to easily test both implementations by yourself.

Hope this helps!
